app/models/spree/variant.rb

```diff
@@ -77,6 +78,7 @@ class Variant < ApplicationRecord
     }
 
     before_validation :set_cost_currency
+    before_validation :ensure_shipping_category
```

Comment:
I'm wondering if the attributes API is better for default values:
```suggestion
    attribute :shipping_category, default: -> { DefaultShippingCategory.find_or_create }
```

Reply:
Hmm interesting, that might mean we can drop some of the other code in the controller too. I'll take a look.

Reply:
Nope, this makes the entire test suite explode :frowning_face: